src/main/java/org/nastya/service/BookService.java

```diff
@@ -138,4 +138,15 @@ public void incrementViewCount(int bookId) throws BookNotFoundException {
         }
         bookViewsDao.incrementViewsCountByBookId(bookId);
     }
+
+    public List<BookListItemDTO> findByTitleContaining(String title) {
+        List<Book> books = bookDao.findByTitleContaining(title);
+        log.info("Found '{}' books containing title '{}'", books.size(), title);
+        List<BookListItemDTO> dtos = bookMapper.mapToBookListItemDTO(books);
+        for (BookListItemDTO dto : dtos) {
```

Comment:
![image](https://github.com/user-attachments/assets/0053d969-30b6-4bc0-978e-175272122b73)

Duplicated code! Please extract the common code into a separate method.